pkg/configuration/load_configuration.go

```diff
@@ -14,48 +14,47 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-// LoadFromSecret retrieves an operator secret and sets environment
-// variables in order to override default configurations.
+// LoadFromSecret retrieves an operator secret and sets a map
+// in order to override default configurations.
 // If no secret is found, then configuration will use defaults.
 // Returns error if WATCH_NAMESPACE is not set, if the resource GET request failed
 // (for other reasons apart from isNotFound) and if setting env vars fails.
 //
 // prefix: represents the operator prefix (HOST_OPERATOR/MEMBER_OPERATOR)
 // resourceKey: is the env var which contains the secret resource name.
-// cl: is the client that should be used to retrieve the configmap.
-func LoadFromSecret(prefix, resourceKey string, cl client.Client) error {
+// cl: is the client that should be used to retrieve the secret.
+func LoadFromSecret(prefix, resourceKey string, cl client.Client) (map[string]string, error) {
+	var secretData = make(map[string]string)
+
 	// get the secret name
 	secretName := getResourceName(resourceKey)
 	if secretName == "" {
-		return nil
+		return secretData, nil
 	}
 
 	// get the secret
 	namespace, err := k8sutil.GetWatchNamespace()
 	if err != nil {
-		return err
+		return secretData, err
 	}
 
 	secret := &v1.Secret{}
 	namespacedName := types.NamespacedName{Namespace: namespace, Name: secretName}
 	err = cl.Get(context.TODO(), namespacedName, secret)
 	if err != nil {
 		if !errs.IsNotFound(err) {
-			return err
+			return secretData, err
 		}
 		logf.Log.Info("secret is not found")
 	}
 
 	// get secrets and set environment variables
 	for key, value := range secret.Data {
 		secretKey := createOperatorEnvVarKey(prefix, key)
-		err := os.Setenv(secretKey, string(value))
-		if err != nil {
-			return err
-		}
+		secretData[secretKey] = string(value)
```

Comment:
btw, do we still need to use the env var format? I mean to have it capitalized, with the prefix and with underscores...?
We don't use it as an env var anymore. 
If we keep it in the original format, then we don't have to do the same transformation here  https://github.com/codeready-toolchain/host-operator/pull/255/files#diff-ec24f891497e9e435f8f9956e23cb379R172 correct?

Reply:
@MatousJobanek good point and good catch! we shouldn't do the transformation unnecessarily, i'll fix this

Reply:
well, I was talking only about the  `secretKey := createOperatorEnvVarKey(prefix, key)` part, but your version works too

Reply:
@MatousJobanek hm, actually this is less efficient. i've changed it here so we only need to convert the value to a string once rather than every time we get the config value: d24d418681e1bcca5e857f042cb5d6f07fa08162